attributes-loader/src/main/scala/it/pagopa/interop/attributesloader/Jobs.scala

```diff
@@ -0,0 +1,134 @@
+    "Societa' in Conto Economico Consolidato",
+    "Stazioni Appaltanti"
+  )
+
+  def loadCertifiedAttributes(bearerToken: String)(implicit
+    contexts: Seq[(String, String)],
+    logger: LoggerTakingImplicit[ContextFieldsToLog],
+    ec: ExecutionContext
+  ): Future[Unit] = {
+    val operationLabel: String = s"Loading certified attributes from Party Registry"
+    logger.info(operationLabel)
+
+    for {
+      categories <- getAllPages(50)((page, limit) =>
+        partyRegistryService.getCategories(Some(page), Some(limit)).map(_.items)
+      )
+
+      attributeSeedsCategoriesNames = categories
+        .map(c =>
+          AttributeSeed(
+            code = Option(c.code),
+            kind = AttributeKind.CERTIFIED,
+            description = c.name, // passing the name since no description exists at party-registry-proxy
+            origin = Option(c.origin),
+            name = c.name
+          )
+        )
+
+      attributeSeedsCategoriesKinds = categories
+        .distinctBy(_.kind)
+        .filterNot(c => kindToBeExcluded.contains(c.kind)) // Including only Pubbliche Amministrazioni
+        .map(c =>
+          AttributeSeed(
+            code = Option(Digester.toSha256(c.kind.getBytes)),
+            kind = AttributeKind.CERTIFIED,
+            description = c.kind,
+            origin = Option(c.origin),
+            name = c.kind
+          )
+        )
+
+      attributeSeedsCategories = attributeSeedsCategoriesKinds ++ attributeSeedsCategoriesNames
+
+      institutions <- getAllPages(50)((page, limit) =>
+        partyRegistryService.getInstitutions(Some(page), Some(limit)).map(_.items)
+      )
+      attributeSeedsInstitutions = institutions.map(i =>
+        AttributeSeed(
+          code = Option(i.originId),
+          kind = AttributeKind.CERTIFIED,
+          description = i.description,
+          origin = Option(i.origin),
+          name = i.description
+        )
+      )
+
+      _ <- addNewAttributes(attributeSeedsCategories ++ attributeSeedsInstitutions)(bearerToken)
+    } yield ()
+  }
+
+  private def addNewAttributes(
+    attributesSeeds: Seq[AttributeSeed]
+  )(bearerToken: String)(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Unit] = {
+
+    // calculating the delta of attributes
+    def delta(attrs: List[PersistentAttribute]): Set[AttributeSeed] =
+      attributesSeeds.foldLeft[Set[AttributeSeed]](Set.empty)((attributesDelta, seed) =>
+        attrs
+          .find(persisted => seed.origin == persisted.origin && seed.code == persisted.code)
+          .fold(attributesDelta + seed)(_ => attributesDelta)
+      )
+
+    // create all new attributes
+    for {
+      attributesfromRM <- getAll(50)(readModelService.find[PersistentAttribute]("attributes", Filters.empty(), _, _))
+      deltaAttributes = delta(attributesfromRM.toList)
+      // The client must log in case of errors
+      _ <- Future.parCollectWithLatch(100)(deltaAttributes.toList)(attributeSeed =>
```

Comment:
this is copied 1:1 from the current logic, I assumed that it was correct.
Should this be addressed?

Reply:
It can be addressed now probably. It depends on what's more important, resiliency or control over the execution. Not a while ago the job was silently failing for all the attributes and it became a silent no-op. Using "non-par" methods will make the chain fail at the first error, but it will give you the chance to log.
Logging and collecting will definitively need another overload in that commons library (that can be absolutely done if it's the preferable way).

Reply:
I agree, it is a required activity, but to me it seems out of the scope of this PR.
I'll open a separated task for this

Reply:
Okay, I can tackle that activity in a binary compatible way in minutes, just send me the task reference

Reply:
[here](https://pagopa.atlassian.net/browse/PIN-3607) it is ❤️